group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1350,6 +1351,7 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
                         .withMembers(group.members())
                         .withStaticMembers(group.staticMembers())
                         .withSubscriptionMetadata(subscriptionMetadata)
+                        .withSubscriptionType(ConsumerGroup.subscriptionType(subscribedTopicNamesMap))
```

Comment:
I was thinking a little more about this one and I still believe that we should rather do it right after computing `subscribedTopicNamesMap = group.computeSubscribedTopicNames(member, updatedMember);` because there are cases where we would end up here but without needing to re-compute it (e.g. staticMemberReplaced is true).